CMakeLists.txt

```diff
@@ -383,10 +383,23 @@ ELSE()
     KOKKOSKERNELS_ADD_TEST_DIRECTORIES(sparse/unit_test)
   ENDIF()
   IF (KokkosKernels_ENABLE_ALL_COMPONENTS)
-    KOKKOSKERNELS_ADD_TEST_DIRECTORIES(perf_test)
-    KOKKOSKERNELS_ADD_EXAMPLE_DIRECTORIES(example)
+    IF (KokkosKernels_ENABLE_PERFTESTS)
+      MESSAGE(STATUS "Enabling perf tests.")
+      KOKKOSKERNELS_ADD_TEST_DIRECTORIES(perf_test)
+    ENDIF ()
+    IF (KokkosKernels_ENABLE_EXAMPLES)
+      MESSAGE(STATUS "Enabling examples.")
+      KOKKOSKERNELS_ADD_EXAMPLE_DIRECTORIES(example)
+    ENDIF ()
   ENDIF()
 
+  IF (NOT KokkosKernels_ENABLE_ALL_COMPONENTS AND KokkosKernels_ENABLE_PERFTESTS)
```

Comment:
Nit picking but an `ELSE` statement above would make sense no?

Reply:
Good point, I made that change.